topdown/crypto.go

```diff
@@ -41,3 +42,67 @@ func builtinCryptoX509ParseCertificates(a ast.Value) (ast.Value, error) {
 func init() {
 	RegisterFunctionalBuiltin1(ast.CryptoX509ParseCertificates.Name, builtinCryptoX509ParseCertificates)
 }
+
+// createRootCAs creates a new Cert Pool from scratch or adds to a copy of System Certs
+func createRootCAs(tlsCaCertFile string, tlsCaCertEnvVar []byte, tlsUseSystemCerts bool) (*x509.CertPool, error) {
+
+	var newRootCAs *x509.CertPool
+
+	if tlsUseSystemCerts {
+		systemCertPool, err := x509.SystemCertPool()
+		if systemCertPool == nil {
+			return nil, err
+		}
+		newRootCAs = systemCertPool
+	} else {
+		newRootCAs = x509.NewCertPool()
+	}
+
+	if len(tlsCaCertFile) > 0 {
+		// Append our cert to the system pool
+		caCert, err := ReadCertFromFile(tlsCaCertFile)
+		if err != nil {
+			return nil, err
+		}
+		if ok := newRootCAs.AppendCertsFromPEM(caCert); !ok {
+			return nil, err
+		}
+	}
+
+	if len(tlsCaCertEnvVar) > 0 {
+		// Append our cert to the system pool
+		if ok := newRootCAs.AppendCertsFromPEM(tlsCaCertEnvVar); !ok {
+			return nil, fmt.Errorf("error appending cert from Env Var into System Certs")
```

Comment:
Let's try to be a bit more helpful, what do you think about this:
```suggestion
			return nil, fmt.Errorf("error appending cert from env var %q into system certs", tlsCaCertEnvVar)
```
I mean, `tlsCaCertEnvVar` is probably huge, so this might be a bad suggestion. I'm trying to anticipate which errors would be more useful when this new feature isn't used properly 🤔 Users make mistakes, and it's sometimes hard to figure out what went wrong... 💭